src/musicbrainz.js

```diff
@@ -38,9 +35,96 @@ exports.track = (name, artist) => {
     }, (response) => {
       var body = "";
       response.on('data', (chunk) => body += chunk)
-      response.on('end', () => resolve(JSON.parse(body)))
+      response.on('end', () => resolve(JSON.parse(body)));
     })
   }).then((json) => {
-    return json.recordings[0]
+      let sortedRecordings = json.recordings.sort((a, b) => a.score - b.score);
+      return sortedRecordings[sortedRecordings.length-1];
   });
 }
+
+
+exports.coverArt = (recording) => {
+    let release = getBestRelease(recording)["release-group"]["id"];
+    if(release) {
+        return new Promise((resolve, reject) => {
+            http.get({
+                protocol: "http:",
+                hostname: "coverartarchive.org",
+                headers: {
+                    "User-Agent": "Application 88nine/0.1.0 (radiomilwaukee.org)",
+                    "Accept": "application/json"
+                },
+                path: `/release-group/${release}`,
+            }, (response) => {
+                if(response.statusCode == 200) {
+                    var body = "";
+                    response.on('data', (chunk) => body += chunk)
+                    response.on('end', () => resolve(JSON.parse(body)))
+                } else {
+                    resolve(null)
+                }
+            })
+        }).then((json) => {
+            return json
+        });
+    } else {
+        return new Promise((resolve, reject) => resolve({}))
+    }
+}
+
+
+function getBestRelease(recording) {
+    // Group releases. If there is a common release group use that.
+    let groupedReleases = recording.releases.reduce((acc, i) => {
+        let rg = i["release-group"];
+        if(rg) {
+            let val = acc[rg.id] || 0;
+            acc[rg.id]++;
+            return acc;
+        }
+    }, {});
+
+    let release = null;
+    let maxLength = 0;
+
+    // Use the most commonly used group.
+    for (var key in groupedReleases ) {
+        let val = groupedReleases[key];
+        if(val > maxLength && val != 1) {
+            release = recording.releases.find((i) => {
+                let rg = i["release-group"];
+                return rg && rg["id"] == key;
+            });
+        }
+    }
+
+    // If we don't have a common release group, find the first official one with type EP
+    if(!release) {
+        release = recording.releases.find(function(i){
+            let rg = i["release-group"];
+            return i["status"] == 'Official' && rg && rg["primary-type"] == "EP";
```

Comment:
I think "Album" is also a primary type that we'd want use?

https://musicbrainz.org/doc/Release_Group/Type